core/src/main/java/org/apache/calcite/rex/RexLiteral.java

```diff
@@ -456,9 +456,13 @@ private static RexDigestIncludeType shouldIncludeType(Comparable value, RelDataT
    * criteria as {@link #valueMatchesType}. */
   public static boolean validConstant(Object o, Litmus litmus) {
     if (o == null
```

Comment:
It seems that this method is only used in tests ?

Reply:
I think so ~